src/main/groovy/com/pszymczyk/consul/ConsulStarter.groovy

```diff
@@ -141,13 +147,17 @@ class ConsulStarter {
             .start()
 
         logHandler.handleStream(innerProcess.getInputStream())
-        ConsulProcess process = new ConsulProcess(dataDir, consulPorts, advertise, innerProcess)
-
+        ConsulProcess process = new ConsulProcess(dataDir, consulPorts, advertise, token, innerProcess)
         logger.info("Starting Consul process on port {}", consulPorts.httpPort)
-        new ConsulWaiter(advertise, consulPorts.httpPort).awaitUntilConsulStarted()
-        logger.info("Consul process started")
-
-        return process
+        try {
```

Comment:
I think we can change the 
` void awaitUntilConsulStarted() {` 
to `boolean awaitUntilConsulStarted() {` 
and remove this try catch clause